sdk/tdf_config.go

```diff
@@ -10,6 +10,8 @@ import (
 const (
 	tdf3KeySize        = 2048
 	defaultSegmentSize = 2 * 1024 * 1024 // 2mb
+	maxSegmentSize     = defaultSegmentSize * 2
```

Comment:
Some motivation, but nothing scientific.  In general we want this value to be as low as possible, and it's trivial to allow a larger value while potentially impossible to reduce.  For that reason I figured this was a good starting place, particularly when I am not aware of any reason to use a larger value outside of less overhead in the segments.

4 MB per decrypting client could still allow for a lot of room for a limited number of clients to use a lot of memory.   So I would recommend only setting this higher if there is a known use case we need to support.

> Following Postel's law, we should probably accept a wider range of values in decrypt than we allow in encrypt.

My take here is that the decrypt limits likely can never be made more restrictive.  We are lucky that it sounds like this is early enough we can still set reasonable limits.  But since this is the initial implementation I believe these can be uniform.  If you prefer I could separate it out to multiple fields to make this more obvious.

Thank you!

Reply:
Ah, I was confused because of the name of the test.